lang/cs/Org.Apache.REEF.Utilities/Collections/ReadOnlySet.cs

```diff
@@ -0,0 +1,137 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace Org.Apache.REEF.Utilities.Collections
+{
+    /// <summary>
+    /// An implementation of a set that is read only.
+    /// Backed by a HashSet.
+    /// </summary>
+    public sealed class ReadOnlySet<T> : IReadOnlyCollection<T>, ISet<T>
+    {
+        private readonly ISet<T> _backingSet;
```

Comment:
This class inherits from ISet and also wraps ISet. You might just want to keep one.